src/query/api/v1/handler/fetch_options.go

```diff
@@ -111,19 +116,39 @@ func (b fetchOptionsBuilder) NewFetchOptions(
 				"could not parse metrics type: input=%s, err=%v", str, err)
 			return nil, xhttp.NewParseError(err, http.StatusBadRequest)
 		}
+
 		fetchOpts.RestrictFetchOptions = newOrExistingRestrictFetchOptions(fetchOpts)
 		fetchOpts.RestrictFetchOptions.MetricsType = mt
 	}
+
 	if str := req.Header.Get(MetricsStoragePolicyHeader); str != "" {
 		sp, err := policy.ParseStoragePolicy(str)
 		if err != nil {
 			err = fmt.Errorf(
 				"could not parse storage policy: input=%s, err=%v", str, err)
 			return nil, xhttp.NewParseError(err, http.StatusBadRequest)
 		}
+
 		fetchOpts.RestrictFetchOptions = newOrExistingRestrictFetchOptions(fetchOpts)
 		fetchOpts.RestrictFetchOptions.StoragePolicy = sp
 	}
+
+	if str := req.Header.Get(FetchRestrictLabels); str != "" {
+		promMatchers, err := promql.ParseMetricSelector(str)
+		if err != nil {
+			return nil, xhttp.NewParseError(err, http.StatusBadRequest)
+		}
+
+		fetchOpts.RestrictFetchOptions = newOrExistingRestrictFetchOptions(fetchOpts)
+		// TODO: filter through TagOptions
```

Comment:
Think this still needs to happen yeah?  Otherwise we'll get wrong tag options being used for these matchers?

Reply:
Yeah I guess; it's only really relevant (atm) if you're overriding the __name__ tag, but will do it here

Reply:
Just as a follow up, this was refactored to no longer be neccessary